### PR TITLE
Alfoa/go on ensemble model

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Hybrid Model

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by alfoa** [comment]
@wangcj05 @mandd

File: tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/test_rom_trainer.xml [hunk 1]
```diff
@@ -148,7 +148,7 @@
     </PointSet>
     <PointSet inputTs="2" name="Pointset_from_database_for_rom_trainer">
       <Input>DeltaTimeScramToAux,DG1recoveryTime</Input>
-      <Output>CladTempThreshold</Output>
+      <Output>CladTempThreshold,time</Output>
```
Comment (mandd) [inline tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/test_rom_trainer.xml]: why does the variable time need to be added?

File: framework/Models/HybridModel.py [hunk 2]
```diff
@@ -653,7 +662,7 @@ def _externalRun(self,inRun, jobHandler):
           if isinstance(evaluation, Runners.Error):
             self.raiseAnError(RuntimeError, "The job identified by "+finishedRun.identifier+" failed!")
           # collect output in temporary data object
-          tempExportDict = self.createExportDictionaryFromFinishedJob(finishedRun, False)
+          tempExportDict = evaluation #self.createExportDictionaryFromFinishedJob(finishedRun, False)
```
Comment (wangcj05) [inline framework/Models/HybridModel.py]: remove the comment?

File: framework/Models/HybridModel.py [hunk 3]
```diff
@@ -680,7 +689,7 @@ def _externalRun(self,inRun, jobHandler):
       if isinstance(evaluation, Runners.Error):
         self.raiseAnError(RuntimeError, "The model "+self.modelInstance.name+" identified by "+finishedRun[0].identifier+" failed!")
       # collect output in temporary data object
-      exportDict = self.modelInstance.createExportDictionaryFromFinishedJob(finishedRun[0], True)
+      exportDict = evaluation #self.modelInstance.createExportDictionaryFromFinishedJob(finishedRun[0], True)
```
Comment (wangcj05) [inline framework/Models/HybridModel.py]: remove the commented line?

File: framework/Models/HybridModel.py [hunk 4]
```diff
@@ -696,17 +705,19 @@ def collectOutput(self,finishedJob,output):
...
       self.tempOutputs['uncollectedJobIds'].pop(jobIndex)
     except ValueError:
       jobIndex = None
     if jobIndex is not None and not useROM:
-      self.collectOutputFromDict(exportDict, self.tempTargetEvaluation)
+      self.tempTargetEvaluation.addRealization(evaluation)
+      #self.collectOutputFromDict(exportDict, self.tempTargetEvaluation)
       self.raiseADebug("ROM is invalid, collect ouptuts of Model with job identifier: {}".format(finishedJob.identifier))
-    Dummy.collectOutput(self, finishedJob, output, options = {'exportDict':exportDict})
+    Dummy.collectOutput(self, finishedJob, output )#, options = {'exportDict':exportDict})
```
Comment (wangcj05) [inline framework/Models/HybridModel.py]: same as before.

**Comment by wangcj05** [comment]
@alfoa I did not see any changes about the out stream, do you have a different branch?

File: tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/test_rom_trainer.xml (hunk 1 above)
Comment (alfoa) [reply tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/test_rom_trainer.xml]: removed...but the test does not work still...

**Comment by alfoa** [comment]
addressed the comments... the raven_running_raven stuff does not work still.

**Comment by wangcj05** [comment]
Do you have the test results?

**Comment by wangcj05** [comment]
Ran 485 tests in 543.1 seconds
361 passed, 47 skipped, 0 pending, 124 FAILED

**Pull request merged**